### PR TITLE
glib: fix static build

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Part of the project to make Qt5 build statically

###### Things done
Made libselinux use musl-fts instead of fts when building statically which fixes glib's static build on staging.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Maybe @alyssais or someone else has a better idea to force pkgsStatic to use musl-fts instead of fts.

### CODE REVIEW

**Comment by alyssais** [comment]
> Maybe @alyssais or someone else has a better idea to force pkgsStatic to use musl-fts instead of fts.

I can actually do one better: #153615 fixes the underlying problem in fts.

File: pkgs/os-specific/linux/libselinux/default.nix [hunk 1]
```diff
@@ -1,10 +1,12 @@
 { lib, stdenv, fetchurl, fetchpatch, buildPackages, pcre, pkg-config, libsepol
 , enablePython ? !stdenv.hostPlatform.isStatic, swig ? null, python3 ? null
-, fts
+, fts, musl-fts
```
Comment (lovesegfault) [inline pkgs/os-specific/linux/libselinux/default.nix:3]: You probably want `musl-fts ? null`?

File: pkgs/os-specific/linux/libselinux/default.nix [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 }:
 
 assert enablePython -> swig != null && python3 != null;
-
+let
+  myFts = if stdenv.hostPlatform.isStatic then musl-fts else fts;
```
Comment (lovesegfault) [inline pkgs/os-specific/linux/libselinux/default.nix:8]: Convention would be

```
fts' = if stdenv.hostPlatform.isStatic then musl-fts else fts;
```

File: pkgs/os-specific/linux/libselinux/default.nix [hunk 3]
```diff
@@ -1,10 +1,12 @@
 { lib, stdenv, fetchurl, fetchpatch, buildPackages, pcre, pkg-config, libsepol
 , enablePython ? !stdenv.hostPlatform.isStatic, swig ? null, python3 ? null
-, fts
+, fts, musl-fts
 }:
 
 assert enablePython -> swig != null && python3 != null;
```
Comment (lovesegfault) [inline pkgs/os-specific/linux/libselinux/default.nix:6]: Add an `assert stdenv.hostPlatform.isStatic -> musl-fts != null;`

**Comment by SCOTT-HAMILTON** [comment]
I'm closing this in favor of https://github.com/NixOS/nixpkgs/pull/153615

**Pull request closed without merging**